### PR TITLE
add array append

### PR DESCRIPTION
**Pull request by subotic**

### CODE REVIEW

**Comment by subotic** [comment]
@adamgfraser the formatting should be good now.

I've run those benchmarks where we added additional collections. Some results are as expected, but some are (at least for me) unexpected.

Benchmarks for the `append` operation is as expected.
```
ChunkAppendBenchmarks.vectorAppend   10000  thrpt   50  6259.027 ± 28.823  ops/s
ChunkAppendBenchmarks.chunkAppend    10000  thrpt   50  5310.220 ±  9.351  ops/s
ChunkAppendBenchmarks.arrayAppend    10000  thrpt   50   125.128 ±  3.718  ops/s
ChunkAppendBenchmarks.listAppend     10000  thrpt   50     5.417 ±  0.011  ops/s
```

I was always under the impression, that a `foldLeft` should run fast on a `List`. So seeing this was (at least for me) unexpected:
```
ChunkFoldBenchmarks.foldChunk    10000  thrpt   50  45423.081 ±  232.411  ops/s
ChunkFoldBenchmarks.foldVector   10000  thrpt   50  41639.197 ± 1722.509  ops/s
ChunkFoldBenchmarks.foldList     10000  thrpt   50  36341.044 ±  427.912  ops/s
ChunkFoldBenchmarks.foldZIO      10000  thrpt   50   2247.632 ±  117.424  ops/s
```

What confuses me is that running the same benchmarks but in another mode (AverageTime) gives a different ordering. 

```
ChunkFoldBenchmarks.foldList      1000  avgt   50   2066.681 ±   75.657  ns/op
ChunkFoldBenchmarks.foldChunk     1000  avgt   50   2234.761 ±    4.346  ns/op
ChunkFoldBenchmarks.foldVector    1000  avgt   50   4547.125 ± 1225.743  ns/op
ChunkFoldBenchmarks.foldZIO       1000  avgt   50  37436.357 ±  145.782  ns/op
```

In conclusion, I need to dig a bit deeper into JMH. Benchmarking is hard :)

**Comment by adamgfraser** [comment]
I think the numbers for fold make sense. The throughput is the inverse of the average time. Yeah, can't beat arrays for iterating over. Going to merge now but would be great to do more work on this.

**Review by adamgfraser — APPROVED** [review]
